### PR TITLE
fix: improve highlight consistency

### PR DESCRIPTION
**Pull request by ObserverOfTime**
The first commit is technically unrelated but I didn't want to separate it.

~~I didn't manage to run the tests locally so I'll update them if they fail.~~ Done.

### CODE REVIEW

**Comment by clason** [comment]
@TravonteD could you look at the CSS changes?

File: runtime/queries/css/highlights.scm [hunk 1]
```diff
@@ -50,37 +63,30 @@
 
 (pseudo_element_selector
   "::"
-  (tag_name) @property)
+  (tag_name) @attribute)
```
Comment (lucario387) [inline runtime/queries/css/highlights.scm:66]: Shouldn't this also be `@tag`?

File: runtime/queries/ruby/highlights.scm [hunk 2]
```diff
@@ -122,7 +122,7 @@
 [
   (class_variable)
   (instance_variable)
-] @label
+] @variable.member
 
 ((identifier) @constant.builtin
   (#match? @constant.builtin "^__(callee|dir|id|method|send|ENCODING|FILE|LINE)__$"))
```
Comment (lucario387) [inline runtime/queries/ruby/highlights.scm:128]: Can you also change this to `any-of` as well? :thinking:

File: runtime/queries/snakemake/highlights.scm [hunk 3]
```diff
@@ -18,7 +18,7 @@
     (_
       (directive
         name:
-          _ @label)))
+          _ @property)))
```
Comment (lucario387) [inline runtime/queries/snakemake/highlights.scm:21]: This looks bad :thinking: 
I think I f'd up something in the formatter. This kind of pattern shouldn't happen

Will fix later

File: runtime/queries/ruby/highlights.scm (hunk 2 above)
Comment (lucario387) [reply runtime/queries/ruby/highlights.scm]: Will find a way to do it :+1:

File: runtime/queries/snakemake/highlights.scm (hunk 3 above)
Comment (lucario387) [reply runtime/queries/snakemake/highlights.scm:21]: #5885

**Comment by clason** [comment]
@ObserverOfTime I rebased your PR on the rebased `main`.

**Comment by ObserverOfTime** [comment]
Oh, whoops. :)

**Comment by clason** [comment]
Well, then you get to rebase yourself ;)

File: runtime/queries/css/highlights.scm (hunk 1 above)
Comment (lucario387) [reply runtime/queries/css/highlights.scm:66]: Agreed

**Comment by lucario387** [comment]
I think `pug` should also be updated too.

**Comment by ObserverOfTime** [comment]
It was. :thinking:

**Comment by clason** [comment]
Can we improve the guidance in `CONTRIBUTING.md` as well?

**Comment by ObserverOfTime** [comment]
If you have something to suggest.

**Comment by clason** [comment]
I was hoping you would...

**Comment by lucario387** [comment]
I think it's okay 🤔

**Comment by clason** [comment]
Merging this now to allow preparing the Great Backport (which will include this, and any other query improvements that are safe for Neovim 0.9).

**Pull request merged**